src/main/java/org/opensearch/flowframework/transport/CreateWorkflowTransportAction.java

```diff
@@ -223,40 +231,73 @@ protected void doExecute(Task task, WorkflowRequest request, ActionListener<Work
-                        ),
-                        ActionListener.wrap(updateResponse -> {
-                            logger.info("updated workflow {} state to {}", request.getWorkflowId(), State.NOT_STARTED.name());
-                            listener.onResponse(new WorkflowResponse(request.getWorkflowId()));
-                        }, exception -> {
-                            String errorMessage = "Failed to update workflow " + request.getWorkflowId() + " in template index";
-                            logger.error(errorMessage, exception);
-                            if (exception instanceof FlowFrameworkException) {
-                                listener.onFailure(exception);
-                            } else {
-                                listener.onFailure(new FlowFrameworkException(errorMessage, ExceptionsHelper.status(exception)));
-                            }
-                        })
-                    );
-                }, exception -> {
-                    String errorMessage = "Failed to update use case template " + request.getWorkflowId();
-                    logger.error(errorMessage, exception);
-                    if (exception instanceof FlowFrameworkException) {
-                        listener.onFailure(exception);
+            // This is an existing workflow (PUT)
+            // Fetch existing entry for time stamps
+            logger.info("Querying existing workflow from global context: {}", workflowId);
+            try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+                client.get(new GetRequest(GLOBAL_CONTEXT_INDEX, workflowId), ActionListener.wrap(getResponse -> {
+                    context.restore();
+                    if (getResponse.isExists()) {
+                        Template existingTemplate = Template.parse(getResponse.getSourceAsString());
+                        // Update existing entry, full document replacement
+                        Template template = new Template.Builder().name(templateWithUser.name())
+                            .description(templateWithUser.description())
+                            .useCase(templateWithUser.useCase())
+                            .templateVersion(templateWithUser.templateVersion())
+                            .compatibilityVersion(templateWithUser.compatibilityVersion())
+                            .workflows(templateWithUser.workflows())
+                            .uiMetadata(templateWithUser.getUiMetadata())
+                            .user(templateWithUser.getUser()) // Should we care about old user here?
+                            .createdTime(existingTemplate.createdTime())
+                            .lastUpdatedTime(Instant.now())
+                            .lastProvisionedTime(existingTemplate.lastProvisionedTime())
+                            .build();
```

Comment:
Can we just not do `Template.Builder().lastUpdatedTime(Instant.now()))` to only update one field?

Reply:
`Template.Builder()` initializes with no defaults.

We could add a new constructor to do `Template.Builder(existingTemplate)`.  WDYT?

Reply:
Created a new builder with template arg. Nice suggestion.